samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,275 @@
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// The context object for this turn.
+        /// <param name="cancellationToken"></param>
+        /// A cancellation token that can be used by other objects or threads to receive notice of cancellation.
+        /// <returns></returns>
+        public override async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await base.OnTurnAsync(turnContext, cancellationToken);
+
+            // Save any state changes that might have occurred during the turn.
+            await _conversationState.SaveChangesAsync(turnContext, false, cancellationToken);
+            await _userState.SaveChangesAsync(turnContext, false, cancellationToken);
+        }
+
+        /// <summary>
+        /// Override this in a derived class to provide logic specific to Message activities, such as the conversational logic.
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var signInLink = await GetSignInLinkAsync(turnContext, cancellationToken).ConfigureAwait(false);
+            if (turnContext.Activity.Text.Contains("login"))
+            {
+                string[] path = { ".", "Resources", "options.json" };
+                var member = await TeamsInfo.GetMemberAsync(turnContext, turnContext.Activity.From.Id, cancellationToken);
+                var initialAdaptiveCard = GetAdaptiveCardWithAuthOptions(path, signInLink, turnContext.Activity.From.Name, member.Id);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(initialAdaptiveCard), cancellationToken);
+            }
+            else if (turnContext.Activity.Text.Contains("PerformSSO"))
+            {
+                string[] striPath = { ".", "Resources", "AdaptiveCardWithSSOInRefresh.json" };
+                var varMember = await TeamsInfo.GetMemberAsync(turnContext, turnContext.Activity.From.Id, cancellationToken);
+                var varInitialAdaptiveCard = GetAdaptiveCardWithAuthOptions(striPath, signInLink, turnContext.Activity.From.Name, varMember.Id);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(varInitialAdaptiveCard), cancellationToken);
+            }
+            else
+            {
+                await turnContext.SendActivityAsync(MessageFactory.Text("Please send 'login' for options"), cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// The OAuth Prompt needs to see the Invoke Activity in order to complete the login process.
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
```

Comment:
No explanation added for these parameters

Reply:
Fixed